compiler/damlc/tests/daml-test-files/RecordUpdate.daml

```diff
@@ -3,8 +3,8 @@
 
 -- @INFO range=43:18-43:42; Evaluate
 -- @INFO range=43:25-43:41; Use const
--- @QUERY-LF .modules[] | .values[] | select(.name_with_type | lf::get_value_name($pkg) == ["p_1_0"]) | .expr.rec_upd | (lf::get_field($pkg) == "x") and (.record.val | lf::get_value_name($pkg) == ["origin"]) and (.update.prim_lit.int64 == "1")
--- @QUERY-LF .modules[] | .values[] | select(.name_with_type | lf::get_value_name($pkg) == ["p_1_2"]) | .expr.rec_upd | (lf::get_field($pkg) == "y") and (.record.rec_upd | (lf::get_field($pkg) == "x") and (.record.val | lf::get_value_name($pkg) == ["origin"]) and (.update.prim_lit.int64 == "1")) and (.update.prim_lit.int64 == "2")
+-- @QUERY-LF .modules[] | .values[] | select(.name_with_type | lf::get_value_name($pkg) == ["p_1_0"]) | .expr.let | (.body.rec_upd | (lf::get_field($pkg) == "x") and (.update.prim_lit.int64 == "1")) and (.bindings[].bound.val | (lf::get_value_name($pkg) == ["origin"]))
```

Comment:
I’m confused. All of this operates on let bindings but your PR switches to case bindings?

Reply:
Yes! Ghc turns the single case binding back into a let binding ...